auth/auth.go

```diff
@@ -90,6 +90,14 @@ func (auth *Authenticator) GetUser(name string) (User, error) {
 
 // Looks up the information for a role.
 func (auth *Authenticator) GetRole(name string) (Role, error) {
+	role, err := auth.GetRoleIncDeleted(name)
+	if role != nil && role.IsDeleted() {
+		return nil, err
+	}
+	return role, err
+}
+
+func (auth *Authenticator) GetRoleIncDeleted(name string) (Role, error) {
```

Comment:
Looks like this can be private.  And since the name isn't immediately obvious, stick a comment in describing how it differs from GetRole.

Reply:
It needs to be public so that it can be used to calculate revocations. This is used in https://github.com/couchbase/sync_gateway/pull/4986/files#diff-c2dab3b01f62851ce6742b56d378463c64d11ed2a1f1d9da5055781e819c3a8cR231

Reply:
If you want I can make them private in this PR as they have no reason not to be in this PR?